src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -3551,11 +3585,22 @@ private void onApplicationDataContOrFin(
                         transferBuilder.more(1);
                     }
 
-                    final AmqpTransferFW transfer = transferBuilder.build();
                     final DirectBuffer fragmentBuffer = messageFragment.buffer();
                     final int fragmentOffset = messageFragment.offset();
                     final int fragmentLimit = messageFragment.limit();
                     final int fragmentSize = fragmentLimit - fragmentOffset;
+                    if (maxMessageSizeExceeded)
+                    {
+                        transferBuilder.aborted(1);
+                        doEncodeTransfer(traceId, authorization, outgoingChannel, transferBuilder.build(),
+                            fragmentBuffer, fragmentOffset, 0);
+                        if (!more)
+                        {
+                            maxMessageSizeExceeded = false;
+                        }
+                        return;
```

Comment:
This code would simplify similarly as above, but we also need to detect when the aggregate size across all `TRANSFER` frames now exceeds the `max-message-size` in total.